bin/createNavData.js

```diff
@@ -1,80 +1,78 @@
 const fse = require('fs-extra');
-const { Observable } = require('rx');
 
-const { commonREs, info, readDir, pagesDir } = require('../utils');
+const { commonREs, info, readDir, loopPages } = require('../utils');
 
 const { isAStubRE, metaTitleRE } = commonREs;
 
 const navData = {};
 
+function readIndex(path) {
+  try {
+    return fse.readFileSync(path, 'utf8');
+  } catch (err) {
+    throw `Could not read file at "${path}".`
```

Comment:
Instead of throwing an error here, could me make the whole build process more stable by inserting a stub and trying again?

We could import `appendStub` from `normaliseArticles` to accomplish this, but we would need to write the file also so that Gatsby doesn't fall over.

Perhaps we could have an article CRUD module to do this for us.

I am open to suggestions, or you telling me it is a bad idea. I haven't had my coffee yet 👍

Reply:
I was going to say yes, but I've changed my mind, I'm going to tell you it's a bad idea 😄 ☕️

This formatted error is mainly meant for the CI. We could add the stub when we run into this, but that would just make the CI absorb the error. The actual Pull Request would still be broken.

Changing my mind again. This is run on build too, so that means this error would be resolved there too, making this less of an issue. But, to keep changing my mind, the stub would still be missing from the repo, which we don't want.

Final verdict: we want to throw an error to make sure the stub is added to the repo.